packages/flutter/test/widgets/draggable_scrollable_sheet_test.dart

```diff
@@ -260,5 +264,48 @@ void main() {
 
       debugDefaultTargetPlatformOverride = null;
     });
+
+    testWidgets('ScrollNotification correctly dispatched when flung without covering its container', (WidgetTester tester) async {
+      final List<Type> notificationTypes = <Type>[];
+      await tester.pumpWidget(_boilerplate(
+        null,
+        onScrollNotification: (ScrollNotification notification) {
+          notificationTypes.add(notification.runtimeType);
+          return false;
+        },
+      ));
+
+      await tester.fling(find.text('Item 1'), const Offset(0, -200), 200);
+      await tester.pumpAndSettle();
+
+      final List<Type> types = <Type>[
+        ScrollStartNotification,
+        ScrollEndNotification,
+      ];
+      expect(notificationTypes, equals(types));
+    });
+
+    testWidgets('ScrollNotification correctly dispatched when flung with contents scroll', (WidgetTester tester) async {
+      final List<Type> notificationTypes = <Type>[];
+      await tester.pumpWidget(_boilerplate(
+        null,
+        onScrollNotification: (ScrollNotification notification) {
+          notificationTypes.add(notification.runtimeType);
+          return false;
+        },
+      ));
+
+      await tester.flingFrom(const Offset(0, 325), const Offset(0, -325), 200);
+      await tester.pumpAndSettle();
+
+      expect(notificationTypes.removeAt(0), ScrollStartNotification);
+      expect(notificationTypes.removeAt(0), UserScrollNotification);
+      expect(notificationTypes.removeLast(), UserScrollNotification);
+      expect(notificationTypes.removeLast(), ScrollEndNotification);
+      expect(
+        notificationTypes.every((Type t) => t == ScrollUpdateNotification),
+        true,
+      );
+    });
```

Comment:
How many notifications are we getting from this?  Can we just assert we get the _exact_ expected number and order of notifications instead of this logic?  Maybe something like:

```dart
final List<Type> expectedNotificationTypes = [
  ScrollStartNotification,
  UserScrollNotification,
  ...List<Type>.filled(expectedNumberOfUpdates, ScrollUpdateNotifcation),
  UserScrollNotification,
  ScrollEndNotification,
];
```

Reply:
Thank you. I fixed test code.